File: exercises/vhavrylov/add_digits/add_digits.py


def add_digits(number):
    digits = []
    while number > 0:
        digits.append(number % 10)
        number = number // 10
    result = 0
    for digit in digits:
        result = result + digit
    if result > 9:
        result = add_digits(result)
    return result

def effective_add_digits(number):
    candidate = number % 9
    if candidate == 0:
        return 9
    return candidate

for i in range (1,1000000):
    #print(i, " ", add_digits(i), " ", i%9)
    assert add_digits(i) == effective_add_digits(i)

File: exercises/vhavrylov/add_digits/add_digits.cpp
// https://leetcode.com/problems/add-digits/

#include <iostream>

using std::cout;
using std::endl;

unsigned int add_digits(unsigned int number) {
  if (number == 0) return 0;
  const unsigned int rest = number % 9;
  return rest ? rest : 9;
}

int main() {
  cout << "Adding digits from 0 to 100" << endl;
  for (unsigned i{0}; i < 100; ++i) {
    cout << i << ":" << add_digits(i) << endl;
  }
  return 0;
}
